utils/python/CIME/XML/component.py

```diff
@@ -22,7 +22,31 @@ def get_value(self, name, attribute={}, resolved=False):
             compnode = self.get_node("components")
```

Comment:
This function confuses me and could use a docstring to explain. The return value can be a string, list, or dictionary depending on name? I'd rather have list, dictionary etc as a <type> than having special cases everywhere.